p2p/conn/secret_connection.go

```diff
@@ -112,6 +114,9 @@ func (sc *SecretConnection) RemotePubKey() crypto.PubKey {
 // Writes encrypted frames of `sealedFrameSize`
 // CONTRACT: data smaller than dataMaxSize is read atomically.
 func (sc *SecretConnection) Write(data []byte) (n int, err error) {
+	// TODO(ismail): is this safe to be called in go-routines?
```

Comment:
I think this should stay unsafe. It's the callers responsibility to serialize access to this connection. See explanations in https://github.com/tendermint/tendermint/issues/3064#issuecomment-453368403 and https://github.com/tendermint/tendermint/issues/2784#issuecomment-453361640